src/jaxsim/rbda/contacts/relaxed_rigid.py

```diff
@@ -360,25 +386,35 @@ def step(carry):
                     A=A,
                     b=b,
                 )
+
                 params = optax.apply_updates(params, updates)
+
                 return params, state
 
-            def continuing_criterion(carry):
+            def continuing_criterion(carry: OptimizationCarry) -> jtp.Bool:
+
                 _, state = carry
+
                 iter_num = optax.tree_utils.tree_get(state, "count")
                 grad = optax.tree_utils.tree_get(state, "grad")
                 err = optax.tree_utils.tree_l2_norm(grad)
+
                 return (iter_num == 0) | ((iter_num < maxiter) & (err >= tol))
 
-            init_carry = (init_params, opt.init(init_params))
             final_params, final_state = jax.lax.while_loop(
                 continuing_criterion, step, init_carry
             )
+
             return final_params, final_state
 
+        # ======================================
+        # Compute the contact forces with L-BFGS
+        # ======================================
+
+        # Initialize the optimized forces with a linear Hunt/Crossley model.
         init_params = (
-            K[:, jnp.newaxis] * jnp.zeros_like(position).at[:, 2].set(δ)
-            + D[:, jnp.newaxis] * velocity
+            K[:, jnp.newaxis] * position_constraint
+            + D[:, jnp.newaxis] * velocity_constraint
```

Comment:
@flferretti this warm start is now computed with the full normal component computed with Hunt/Crossley model. Before, only the z component was used, that was true only on flat terrain.

Reply:
Thanks Diego, you're right! This might be substituted with a different warm-start anyway